kernel/src/table_changes/scan.rs

```diff
@@ -29,8 +33,6 @@ pub struct TableChangesScan {
     predicate: Option<ExpressionRef>,
     // The [`ColumnType`] of all the fields in the `logical_schema`
     all_fields: Vec<ColumnType>,
-    // `true` if any column in the `logical_schema` is a partition column
-    have_partition_cols: bool,
```

Comment:
`have_partition_cols` was useful in scan to skip building the physical to logical expression. [Source](https://github.com/delta-io/delta-kernel-rs/blob/main/kernel/src/scan/mod.rs#L482). For CDF, we always have to construct an expression because it changes based on the scan file.